services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/logging/DisclosureLogService.kt

```diff
@@ -165,24 +167,26 @@ class DisclosureLogService(private val auditLogService: AuditLogService) {
             .flatMap { auditLogEntriesForContacts(it.id!!, it.applicationData, objectType) }
             .toSet()
 
-    private fun extractContacts(applicationData: ApplicationData): List<Contact> =
+    private fun extractContacts(applicationData: ApplicationData): List<ContactWithRole> =
         when (applicationData) {
             is CableReportApplicationData ->
                 applicationData
-                    .customersWithContacts()
-                    .flatMap { it.contacts }
-                    .filter { it.hasInformation() }
+                    .customersByRole()
+                    .flatMap { (role, customer) ->
+                        customer.contacts.map { ContactWithRole(role, it) }
+                    }
+                    .filter { it.contact.hasInformation() }
         }
 
-    private fun extractCustomers(applicationData: ApplicationData): List<Customer> =
+    private fun extractCustomers(applicationData: ApplicationData): List<CustomerWithRole> =
         when (applicationData) {
             is CableReportApplicationData ->
                 applicationData
-                    .customersWithContacts()
-                    .map { it.customer }
+                    .customersByRole()
+                    .map { (role, customer) -> CustomerWithRole(role, customer.customer) }
                     // Only personal data needs to be logged, not other types of customers.
-                    .filter { it.type == CustomerType.PERSON }
-                    .filter { it.hasPersonalInformation() }
+                    .filter { it.customer.type == CustomerType.PERSON }
```

Comment:
I think the comment above is redundant.

`.filter { it.customer.type == CustomerType.PERSON }` is pretty clear on what we are interested in here.

This is just a suggestion / opinion which you can ignore.